syntax = "proto3";

service add {
    rpc add(addMsg) returns (addReply) {}
}

service dotProduct {
    rpc dotProduct(dotProductMsg) returns (dotProductReply) {}
}

service image {
    rpc image(rawImageMsg) returns (imageReply) {}
}

service jsonImage {
    rpc jsonImage(jsonImageMsg) returns (imageReply) {}
}

message addMsg {
  int32 a = 1;
  int32 b = 2;
}

message addReply {
    int32 sum = 1;
}

message dotProductMsg {
    repeated float a = 1;
    repeated float b = 2;
  }

message dotProductReply {
    float dotproduct = 1;
}

message rawImageMsg {
    bytes img=1;
}

message jsonImageMsg {
      string img=1;
}

message imageReply {
    int32 width=1;
    int32 height=2;
}
